server/src/main/java/org/apache/druid/segment/realtime/plumber/Sink.java

```diff
@@ -148,7 +159,9 @@ public Sink(
       maxCount = hydrant.getCount();
       ReferenceCountingSegment segment = hydrant.getIncrementedSegment();
       try {
-        numRowsExcludingCurrIndex.addAndGet(segment.asQueryableIndex().getNumRows());
+        QueryableIndex index = segment.asQueryableIndex();
```

Comment:
Sink is also used for both realtime and batch. How do your changes affect batch ingestion ?

Reply:
Additionally, I am maintaining set of columns in each sink. I am assuming this wouldn't be much of an overhead in batch ingestion.